orthoinference/src/test/java/org/reactome/orthoinference/StableIdentifierGeneratorTest.java

```diff
@@ -0,0 +1,67 @@
+package org.reactome.orthoinference;
+
+import org.gk.model.GKInstance;
+import org.gk.persistence.MySQLAdaptor;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PowerMockIgnore;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.Assert.*;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({StableIdentifierGenerator.class, InstanceUtilities.class})
+@PowerMockIgnore({"org.apache.logging.log4j.*", "javax.management.*", "javax.script.*",
+        "javax.xml.*", "com.sun.org.apache.xerces.*", "org.xml.sax.*", "com.sun.xml.*", "org.w3c.dom.*", "org.mockito.*"})
+public class StableIdentifierGeneratorTest {
+
+    private static Object identifier = "R-HSA-123456";
+    private static String fakeAbbreviation = "ABC";
+    @Mock
+    Map<String,Integer> fakeSeenOrthoIds = new HashMap<>();
+
+    @Mock
+    MySQLAdaptor mockAdaptor;
+
+    @Mock
+    GKInstance mockInferredInst;
+
+    @Mock
+    GKInstance mockOriginalInst;
+
+    @Mock
+    GKInstance mockStableIdentifierInst;
+
+    @Mock
+    GKInstance mockOrthoStableIdentifierInst;
+
+    @Before
+    public void setUp() throws Exception {
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @Test
+    public void generateOrthologousStableIdTest() throws Exception {
+
+        PowerMockito.mockStatic(InstanceUtilities.class);
+        Mockito.when(mockOriginalInst.getAttributeValue("stableIdentifier")).thenReturn(mockStableIdentifierInst);
+        Mockito.when(mockStableIdentifierInst.getAttributeValue("identifier")).thenReturn(identifier);
+
+        StableIdentifierGenerator.setSpeciesAbbreviation(fakeAbbreviation);
+        StableIdentifierGenerator.setAdaptor(mockAdaptor);
+
+        PowerMockito.when(InstanceUtilities.createNewInferredGKInstance(mockStableIdentifierInst)).thenReturn(mockOrthoStableIdentifierInst);
+
+        StableIdentifierGenerator.generateOrthologousStableId(mockInferredInst, mockOriginalInst);
```

Comment:
Can the return value of this call be captured and tested it was updated correctly with an assert?

Reply:
Right, I will do that. I need to kill this coverage priority now before it turns into a bad habit

Reply:
So when testing that the 'stableIdentifier' attribute exists in the inferred instance it returns null, both in the actual code and in the test code as well. I think that since it's a mock GKInstance, that it never actually gets populated. I could get around this by either making it a real GKInstance or maybe by populating the mock instance ahead of time in the test code (which would not prove anything actually worked in the code). Suggestions?